concurrency/MpmcQueue.h

```diff
@@ -249,13 +282,21 @@ class MpmcQueue
 
     // Layout below must be maintained for performance. The non atomic member variables may not be place before read
     // cursor.
-    std::atomic<VersionedIndex> _readCursor;
-    uint32_t _cacheLineSeparator1[14];
-    std::atomic<VersionedIndex> _writeCursor;
-    uint32_t _cacheLineSeparator2[14];
-    const uint32_t _entriesPerCacheline;
-    const uint32_t _capacity;
-    const size_t _blockSize;
+    alignas(64) std::atomic<VersionedIndex> _readCursor;
+
+    // Special Cell state for when queue has zero size
+    // Null Entry is not written and it is only read it when queue is zero size. So it can live in same cache line of
+    // _readCursor without performance impact
+    const EntrySate _nullEntry = CellState::State::nullSlot;
```

Comment:
This is in the same cache line of _readCursor to save a little of memory. This value is unchangeable   and it only read when _readCursor is also unchangeable (ie, when the capacity is zero)